materialize-kafka/src/transactor.rs

```diff
@@ -0,0 +1,211 @@
+use anyhow::{Context, Result};
+use avro::{encode, encode_key};
+use bytes::Bytes;
+use proto_flow::materialize::{
+    request::{Open, Store},
+    response::{Acknowledged, Flushed, Opened, StartedCommit},
+    Response,
+};
+use rdkafka::{
+    error::KafkaError,
+    producer::{BaseRecord, Producer},
+    types::RDKafkaErrorCode,
+    util::Timeout,
+};
+use tuple::Element;
+
+use crate::{
+    binding_info::{get_binding_info, BindingInfo},
+    configuration::{EndpointConfig, MessageFormat},
+    Input, Output,
+};
+
+pub async fn run_transactions(mut input: Input, mut output: Output, open: Open) -> Result<()> {
```

Comment:
I imagine we want to extract this into a library at some point for writing Rust connectors? this seems like `materialize-boilerplate`, but for Rust

Reply:
Yep for sure. Probably with the next Rust materialization we write, whenever that is, especially if it needs to do standard updates.